### PR TITLE
A maze command

### PR DESCRIPTION
**Pull request by Crenshaw7860**
Generates a maze with an optional amount bridges (path parts you can go under/over) I think it would be a nice addition. I hope you enjoy it.

### CODE REVIEW

File: fun/maze.cc.lau [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{{$embed:=sdict "thumbnail" (sdict "url" (.User.AvatarURL "128")) "title" (title "maze") "description" (joinStr "" "Requested by " .User.String " -") "image" (sdict "url" nil) "color" 123}}
+ 
+{{if .CmdArgs}}
+     {{$cross:=(index .CmdArgs 0)}}
+     {{$maze:=(joinStr "" "http://maze5.de/cgi-bin/maze?sample=1&type=4&rows=12&columns=12&crossings=" (index .CmdArgs 0) "&seed=" $seed "&algorithm=backtracker&algorithm=0.5&foreground=%23ffffff&background=%2336393f&bordersize=16&cellsize=32&linewidth=2.5&format=png")}}
+    {{$embed.image.Set "url" $maze}}
+    {{$maze:= (joinStr "" $maze "&solution=true")}}
+    {{$embed.Set "description" (print $embed.description " [Solution](" $maze ")")}}
+ 
+{{else if .ExecData}}
```
Comment (Crenshaw7860) [inline fun/maze.cc.lau:16]: I plan on making this (.ExecData) go away by having a variable set for (index .CmdArgs 0), so I can just have
```
{{$data:=""}}
{{if .ExecData}}
     {{$data = .ExecData}}
{{else}}
     {{$data = (index .CmdArgs 0)}}
{{end}}
```

File: fun/maze.cc.lau [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{{/*Made by Crenshaw#7860
+Based off Raffael#7777's Carl-Bot command here: https://carl.gg/t/260353
+Command type: prefix/mention
+It also shows the solution, and works with .ExecData*/}}
```
Comment (Jo3-L) [inline fun/maze.cc.lau:4]: This comment is formatted incorrectly. Please refer to the contributing guide for how to format the top comment.

File: fun/maze.cc.lau [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ 
+{{$seed:=randInt 100000000 999999999}}
+{{$embed:=sdict "thumbnail" (sdict "url" (.User.AvatarURL "128")) "title" (title "maze") "description" (joinStr "" "Requested by " .User.String " -") "image" (sdict "url" nil) "color" 123}}
+ 
+{{if .CmdArgs}}
+     {{$cross:=(index .CmdArgs 0)}}
+     {{$maze:=(joinStr "" "http://maze5.de/cgi-bin/maze?sample=1&type=4&rows=12&columns=12&crossings=" (index .CmdArgs 0) "&seed=" $seed "&algorithm=backtracker&algorithm=0.5&foreground=%23ffffff&background=%2336393f&bordersize=16&cellsize=32&linewidth=2.5&format=png")}}
```
Comment (Jo3-L) [inline fun/maze.cc.lau:11]: You can use `$maze` here instead of `(index .CmdArgs 0)`.

**Review by Jo3-L — CHANGES_REQUESTED** [review]
Just a few things to change, otherwise LGTM.

**Comment by Jo3-L** [comment]
thanks for the PR!

**Pull request merged**